### PR TITLE
[CTCP-4898] added identification page for departureMeansOfTransport

### PR DESCRIPTION
**Pull request by prashanthmamidi**

### CODE REVIEW

**Comment by prashanthmamidi** [comment]
image

**Comment by prashanthmamidi** [comment]
image

**Comment by prashanthmamidi** [comment]
image

File: conf/messages.en [hunk 1]
```diff
@@ -1114,3 +1114,12 @@ houseConsignment.index.additionalReference.addAdditionalReferenceNumberYesNo.hea
 houseConsignment.index.additionalReference.addAdditionalReferenceNumberYesNo.error.required = Select yes if you want to add an additional reference number
 
 unloadingFindings.rowHeadings.houseConsignment.countryOfDestination = Country of destination
+
+houseConsignment.index.departureMeansOfTransport.identification.NormalMode.title = Which identification do you want to use for the departure means of transport?
+houseConsignment.index.departureMeansOfTransport.identification.CheckMode.title = Which identification do you want to use for the new departure means of transport?
```
Comment (justega247) [inline conf/messages.en:1119]: This title needs to be updated.

File: conf/messages.en [hunk 2]
```diff
@@ -1114,3 +1114,12 @@ houseConsignment.index.additionalReference.addAdditionalReferenceNumberYesNo.hea
...
+houseConsignment.index.departureMeansOfTransport.identification.NormalMode.heading = Which identification do you want to use for the departure means of transport?
+houseConsignment.index.departureMeansOfTransport.identification.CheckMode.heading = Which identification do you want to use for the new departure means of transport?
```
Comment (justega247) [inline conf/messages.en:1121]: This heading needs to be updated.

File: test/viewModels/houseConsignment/index/departureTransportMeans/IdentificationViewModelSpec.scala [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      result.paragraph mustBe None
+    }
+
+    "when CheckMode" in {
+      val viewModelProvider = new IdentificationViewModelProvider()
+
+      val result = viewModelProvider.apply(CheckMode)(messages)
+
+      result.title mustBe "Which identification do you want to use for the new departure means of transport?"
+      result.heading mustBe "Which identification do you want to use for the new departure means of transport?"
```
Comment (justega247) [inline test/viewModels/houseConsignment/index/departureTransportMeans/IdentificationViewModelSpec.scala:43]: These values need to be updated.

**Comment by prashanthmamidi** [comment]
image

**Comment by prashanthmamidi** [comment]
image

File: conf/messages.en [hunk 4]
```diff
@@ -1117,8 +1117,9 @@ unloadingFindings.rowHeadings.houseConsignment.countryOfDestination = Country of
 
 houseConsignment.index.departureMeansOfTransport.identification.NormalMode.title = Which identification do you want to use for the departure means of transport?
 houseConsignment.index.departureMeansOfTransport.identification.CheckMode.title = Which identification do you want to use for the new departure means of transport?
```
Comment (justega247) [inline conf/messages.en:1119]: Does this line then need to be deleted?

Comment (prashanthmamidi) [reply conf/messages.en]: yes

**Review by justega247 — DISMISSED** [review]

**Review by justega247 — APPROVED** [review]

File: app/views/houseConsignment/index/departureMeansOfTransport/IdentificationView.scala.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      if (viewModel.paragraph.isDefined)
+           RadioWithAdditionalHtml(
+               caption = Some(s"MRN: $mrn"),
+               heading = viewModel.heading,
+               additionalHtml = viewModel.paragraph.map(_.toParagraph).get)
+      else
+          Radio(caption = Some(s"MRN: $mrn"), heading = viewModel.heading),
+
+         radioItems = fieldId => identifiers.toRadioItems(fieldId, form.value)
+      )
```
Comment (jackheslop96) [inline app/views/houseConsignment/index/departureMeansOfTransport/IdentificationView.scala.html:55]: Could have an apply method in `InputRadioViewModel`, similar to https://github.com/hmrc/manage-transit-movements-unloading-frontend/blob/42ada085e26e8f9ebb28d295021cfb02305801c0/app/viewModels/components/InputYesNoViewModel.scala#L42-L49

**Pull request merged**